core/src/main/java/org/axonframework/commandhandling/model/inspection/AnnotatedAggregate.java

```diff
@@ -213,6 +307,21 @@ protected boolean getIsLive() {
         return true;
     }
 
+    @Override
+    protected <R> Aggregate<R> doCreateNew(Class<R> aggregateType, Callable<R> factoryMethod) throws Exception {
+        if (repositoryProvider == null) {
+            throw new AxonConfigurationException(format(
```

Comment:
Maybe this is still to early in the PR, so I haven't spotted it yet, but still; 
Is this a desirable situation? 
What will be the default, no `RepositoryProvider` or some form of default provider? 

I assume this is necessary because the earlier classes I commented on did not enforce setting the `RepositoryProvider`, correct?
Why wouldn't we want to enforce setting such a provider?